scripts/histmakers/3dsmoothing/combinedifferentbinning.C

```diff
@@ -0,0 +1,86 @@
+//TO BE USED IN CASE WE NEED TO COMBINE EFFICIENCIES SMOOTHED IN DIFFERENT PT RANGES (AT THE MOMENT THE SMOOTHER DOESN'T DO IT, SO A WORKAROUND IS TO CHANGE THE PT RANGE IN THE HISTOGRAMS AND THEN ADD THE REMAINING BINS FROM ANOTHER FILE).
+
+void combinedifferentbinning() {
+	TFile *file=new TFile("/gpfs/ddn/cms/user/bruschin/Wsmooth/smoothLeptonScaleFactorsNoScetlib/GtoH/allSmooth_GtoH.root");
+	TFile *file2=new TFile("/gpfs/ddn/cms/user/bruschin/Wsmooth/smoothLeptonScaleFactors20bins/GtoH/allSmooth_GtoH.root");
+	TFile *file3=new TFile("/gpfs/ddn/cms/user/bruschin/Wsmooth/smoothLeptonScaleFactors20bins/GtoH/allSmooth_GtoH2.root","RECREATE");
+	file3->cd();
+	for (auto&& keyAsObj : *(file->GetListOfKeys())){
+		auto key = (TKey*) keyAsObj;
+		TH2D *_2dhisto1, *_2dhisto2, *_2dhisto3;
+		TH3D *_3dhisto1, *_3dhisto2, *_3dhisto3;
+		double *xaxis, *yaxis, *zaxis;
+		if (std::string(key->GetClassName())==std::string("TH2D")) {
+			std::string name(key->GetName());
+			_2dhisto1=(TH2D*)file->Get(name.c_str());
+			_2dhisto1->SetName("histo1");
+			_2dhisto2=(TH2D*)file2->Get(name.c_str());
+			_2dhisto2->SetName("histo2");
+			xaxis=(double*)malloc((_2dhisto1->GetXaxis()->GetNbins()+1)*sizeof(double));
```

Comment:
I would suggest using std::vector instead of malloc.
Also in the loop to take the bin edges you can still use GetBinLowEdge reading the index for the overflow bin, in this way you can do everything inside the same loop